web/src/test/java/com/epam/rft/atsy/web/controllers/WelcomeControllerTest.java

```diff
@@ -1,28 +1,49 @@
 package com.epam.rft.atsy.web.controllers;
 
-import static org.hamcrest.CoreMatchers.is;
-import static org.hamcrest.MatcherAssert.assertThat;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.forwardedUrl;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.view;
 
-import org.mockito.InjectMocks;
-import org.mockito.MockitoAnnotations;
-import org.springframework.web.servlet.ModelAndView;
-import org.testng.annotations.BeforeMethod;
-import org.testng.annotations.Test;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.servlet.ViewResolver;
+import org.springframework.web.servlet.view.InternalResourceViewResolver;
 
 public class WelcomeControllerTest {
-  @InjectMocks
-  private WelcomeController underTest;
+  private static final String VIEW_PREFIX = "/WEB-INF/pages/";
+  private static final String VIEW_SUFFIX = ".jsp";
 
-  @BeforeMethod
+  private static final String VIEW_NAME = "candidates";
+
+  private static final String REQUEST_URL = "/secure/welcome";
+
+  private MockMvc mockMvc;
+
+  @Before
   public void setUp() {
```

Comment:
This setup method would look like almost the same for all Spring MockMvc based tests, so this means that it can be reused.

As a first thought I would create an AbstractMockMvcTest class, which will contain the setup logic, the mockMvc field and the definition of the viewresolver. All of them as protected, so the extending classes can overwrite them if needed.

Also to control which controllers are set up by the mock mvc, you can use the template method design pattern for the current setup method. Ie: define abstract methods which return the list of controller instances.